### PR TITLE
Code Quality: Introducing StandardViewBase to reduce code duplication

### PR DESCRIPTION
**Pull request by QuaintMako**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Related #10305 

**What is the purpose of that PR.**
A lot of code duplication can be found inside the ViewBrowsers. This PR aims at reducing the duplication by combining the concerned classes inside a standard view. 

This preliminary work has for objective to pave the way to a tackle #1928 further down the line.

**What has been done**
- Introducing `StandardViewBase` to gather up the duplicated code in `GridViewBrowser`, `ColumnViewBase` and `GridViewBrowser`.

**Validation**
How did you test these changes?
- [X] Built and ran the app.

### CODE REVIEW

**Comment by QuaintMako** [comment]
Setting this up as a draft so discussions may ensue to make sure it goes into the right direction.

**Comment by yaira2** [comment]
@lukeblevins

**Comment by yaira2** [comment]
@QuaintMako what are your plans for the current BaseLayout, will that be removed?

**Comment by QuaintMako** [comment]
> @QuaintMako what are your plans for the current BaseLayout, will that be removed?

What I aim with this PR is to reduce the duplication inside `GridViewBrowser`, `ColumnViewBase` and `GridViewBrowser` who share a pretty close philosophy. They do inherit the `BaseLayout`, but so does `ColumnViewBrowser` who is completely different to the others. That's why I introduced something between `BaseLayout` and the three classes I am refactoring.

For now there is no plan around the BaseLayout. But looking at it, we may have ways to simplify it.

**Comment by QuaintMako** [comment]
Opening up this PR for a first refactoring of the different views.
When it's done and merged, I'll look into defining another object between the Grid and Details layouts to reduce even more the duplication of the code.

File: src/Files.App/BaseLayout.cs [hunk 1]
```diff
@@ -248,7 +248,6 @@ internal set
 					}
 
 					NotifyPropertyChanged(nameof(SelectedItems));
-					//ItemManipulationModel.SetDragModeForItems();
```
Comment (yaira2) [inline src/Files.App/BaseLayout.cs:251]: Why is this commented out?

**Review by yaira2 — APPROVED** [review]
LGTM

File: src/Files.App/BaseLayout.cs (hunk 1 above)
Comment (QuaintMako) [reply src/Files.App/BaseLayout.cs:251]: No idea, it has been introduced in #5358 but I don't know why it has been.

Comment (yaira2) [reply src/Files.App/BaseLayout.cs:251]: Can you remove if it's not needed?

Comment (QuaintMako) [reply src/Files.App/BaseLayout.cs:251]: You mean at it back the commented line?

Comment (yaira2) [reply src/Files.App/BaseLayout.cs:251]: Remove the whole line

File: src/Files.App/Views/LayoutModes/StandardLayoutMode.cs [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+			}
+		}
+
+		protected override void Page_CharacterReceived(UIElement sender, CharacterReceivedRoutedEventArgs args)
+		{
+			if (ParentShellPageInstance is null)
+				return;
+
+			if (ParentShellPageInstance.CurrentPageType != this.GetType() || IsRenamingItem)
+				return;
```
Comment (ferrariofilippo) [inline src/Files.App/Views/LayoutModes/StandardLayoutMode.cs:220]: You may merge these two ifs to reduce branching

File: src/Files.App/BaseLayout.cs (hunk 1 above)
Comment (yaira2) [reply src/Files.App/BaseLayout.cs:251]: Lol, I didn't see the red.

Comment (QuaintMako) [reply src/Files.App/BaseLayout.cs:251]: No problem!

**Review by ferrariofilippo — APPROVED** [review]
LGTM

**Pull request merged**